### DISCUSSION TITLE
UUID in rez request?

### DISCUSSION THREAD

**Discussion by hughetop** [AcademySoftwareFoundation/rez#1370] [Q&A] [answered]
Hello,

I am running into some issues with a package (unfortunately) named 'maya', which is conflated with the software of the same name. I can utilize the `uuid` field in package.py to distinguish between the two during build and install, but how do I request one over the other? Similar to the [hello_world example](https://github.com/AcademySoftwareFoundation/rez/blob/master/example_packages/hello_world/package.py#L22), I put `uuid = 'studio.maya'` in the package.py, but is there a way to request that package based on uuid? I tried `rez-env studio.maya` but that gave a PackageFamilyNotFoundError.

So is there a way to distinguish packages with the same name during a rez-env?

**Comment by JeanChristopheMorinPerso** [comment]
I don't think that's possible, but maybe I'm wrong. I'll let some time for anybody else to answer before I actually take a closer look at the code.

**Reply by hughetop** [reply]
Yeah, I assumed so. I couldn't find anything in the docs or wiki suggesting how to distinguish between the two packages during a rez-env. We will likely just have to change the name of the custom package and deal with refactoring.

**Comment by nerdvegas** [comment]
This isn't possible, by design. The (poorly named) uuid is there only to
ensure they you don't accidentally release a package 'foo' into the same
package repo containing an unrelated package 'foo'.

**Reply by hughetop** [reply]
Yeah, fortunately it's a custom package so we're just going to rename it and deal with a big refactor. But if there are multiple 3rd party packages with the same name, is it possible to distinguish them during a rez-env?

**Comment by nerdvegas** [comment, marked as answer]
No, the name _is_ what distinguishes packages (ultimately something has
to!). There could be an argument for allowing a package request to be
qualified with the repo it belongs to, which might help a bit, but I think
that's dangerous (and would be pretty verbose).

I don't have a good answer for this, it's not something that's come up
much. But, if rez usage expanded a lot and started managing more packages
from more ecosystems, I could see name clashes becoming more of an issue.

Something to think about,

**Reply by hughetop** [reply]
Yeah, fortunately it's a custom package and we can rename and refactor. 

For some background, this custom package was deployed under a "studio" namespace, so it was imported like `import studio.maya` or `from studio import maya`, so there wasn't a conflict. We eventually figured out how to do this with rez, by having a "studio" directory with no `__init__.py` as the parent directory. This worked fine, we could still import it "from studio", and it didn't conflict with the software. However, we realized that we can't distinguish the two during a rez-env. Again, we can rename and refactor because it's a custom package, but I do wonder how we would handle this if it were a 3rd party package.
